### ISSUE TITLE
tsql: ALTER TABLE … ENABLE/DISABLE TRIGGER … fails to parse

### ISSUE THREAD

**Issue by msergein** [sqlfluff/sqlfluff#7804]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In the `tsql` dialect, the canonical Microsoft T-SQL statement
`ALTER TABLE <table> { ENABLE | DISABLE } TRIGGER <trigger_name>` does not
parse — sqlfluff reports the line as an unparsable section.

Reference: https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-table-transact-sql
(the `{ ENABLE | DISABLE } TRIGGER { ALL | trigger_name [ ,...n ] }` clause)

### Root cause (from reading the source) by claude code

In `src/sqlfluff/dialects/dialect_tsql.py`, the `AlterTableSegment` does not
contain a branch matching the `{ ENABLE | DISABLE } TRIGGER …` clause. There
are existing standalone segments for similar trigger-related statements
(e.g. `EnableDisableTriggerStatementSegment` exists in some other dialects),
but no equivalent is wired into `tsql`.

Suggested fix: add an alternative inside `AlterTableSegment` (or a separate
`AlterTableTriggerStatementSegment` reachable from `StatementSegment`) along
the lines of:

```python
Sequence(
    OneOf("ENABLE", "DISABLE"),
    "TRIGGER",
    OneOf(
        "ALL",
        Delimited(Ref("TriggerReferenceSegment")),
    ),
)
```

…and a YAML test fixture under
`test/fixtures/dialects/tsql/alter_table_enable_disable_trigger.{sql,yml}`

### Expected Behaviour

All of the following statements should parse cleanly under `--dialect tsql`:

```sql
ALTER TABLE [dbo].[Contact] DISABLE TRIGGER [TRContactID];
ALTER TABLE [dbo].[Contact] ENABLE TRIGGER [TRContactID];
ALTER TABLE [dbo].[Contact] DISABLE TRIGGER ALL;
ALTER TABLE [dbo].[Contact] ENABLE TRIGGER ALL;
ALTER TABLE [dbo].[Contact] ENABLE TRIGGER [t1], [t2];
```

### Observed Behaviour

```
$ cat repro.sql
ALTER TABLE [dbo].[Contact] DISABLE TRIGGER [TRContactID]

$ sqlfluff parse --dialect tsql repro.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: "<Sequence: [<AnyNumberOf: [<Ref: 'BatchSegment'>]>]>"
...
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'ALTER TABLE
                       | [dbo].[Contact] DISABLE TRIGGER [TRContactID]...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

The error happens both on its own and after a preceding `CREATE TRIGGER … END` block (with or without an intervening `GO`).

### How to reproduce

1. Save the following as `repro.sql`:

   ```sql
   ALTER TABLE [dbo].[Contact] DISABLE TRIGGER [TRContactID];
   ```

2. Run:

   ```
   sqlfluff parse --dialect tsql repro.sql
   ```

3. Observe the unparsable section starting at line 1.

### Dialect

tsql 

### Version

4.1.0
Python 3.14.4 (Windows)

### Configuration

[sqlfluff]
dialect = tsql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
